app/routes/_gcn.circulars/circulars.server.ts

```diff
@@ -261,3 +304,169 @@ export async function circularRedirect(query: string) {
     throw redirect(circularURL)
   }
 }
+
+/**
+ * Gets all entries in circulars_history for a given circularId
+ * @param circularId
+ * @returns an array of previous versions of a Circular sorted by version
+ */
+export async function getVersions(circularId: number): Promise<Circular[]> {
+  const db = await tables()
+  const result = await db.circulars_history.query({
+    KeyConditionExpression: 'circularId = :circularId',
+    ExpressionAttributeValues: {
+      ':circularId': circularId,
+    },
+  })
+  return result.Items
+}
```

Comment:
Do we want to return the _content_ of all of the circulars, or just a list of their version numbers?

Also, this does not include the newest or current version number.

Reply:
It should include all versions that exist, since the auto increment puts into both tables, right? I'll need to test a bit. I think in the previous iteration I assumed that the history didn't contain the latest, and I just joined the history and the current entry in the circulars table. I can edit this to make sure it returns the whole list

Reply:
Remember that in @nasa-gcn/dynamodb-autoincrement we landed on a design in which all versions _except for the most recent one_ are stored in the history table.

I suggest the following:

- Change the signature of this function so that it returns a list of version numbers.
- Add a projection to the query so that it only returns the circularId attribute.
- Before returning, sort the versions, and push one additional implied version.

Reply:
Ah right, okay I updated this and another bug in the get method in this latest commit